modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -0,0 +1,212 @@
+
+import java.util.Map;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+/**
+ * Test class of UserResourceManager.
+ */
+@PrepareForTest({AbstractResourceManager.class})
+public class UserResourceManagerTest extends PowerMockTestCase {
+
+    UserResourceManager userResourceManager = new UserResourceManager();
+
+    private User getNewUser() throws BadRequestException, CharonException, InternalErrorException {
+
+        String scimObjectString = "{\n" +
+                "  \"schemas\": \n" +
+                "    [\"urn:ietf:params:scim:schemas:core:2.0:User\",\n" +
+                "    \"urn:ietf:params:scim:schemas:extension:enterprise:2.0:User\"\n" +
+                "    ],\n" +
+                " \"name\": {\n" +
+                "    \"givenName\": \"Rash\",\n" +
+                "    \"familyName\": \"Na\",\n" +
+                "    \"formatted\": \"Rash Na\"\n" +
+                "  },\n" +
+                "  \"userName\": \"rash\",\n" +
+                "  \"password\": \"rash123\",\n" +
+                "  \"id\": \"123\",\n" +
+                "  \"emails\": [\n" +
+                "      {\n" +
+                "        \"type\": \"home\",\n" +
+                "        \"value\": \"rash@gmail.com\",\n" +
+                "         \"primary\": true\n" +
+                "      }\n" +
+                "  ]\n" +
+                "}";
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        User user = (User) decoder.decodeResource(scimObjectString, schema, new User());
+
+        return user;
+    }
+
+    @DataProvider(name = "dataForGetSuccess")
+    public Object[][] dataToGetSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        UserManager userManager = mock(UserManager.class);
+
+        return new Object[][]{
+
+                {id, userManager, "userName", null, 200, user}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetSuccess")
+    public void testGetSuccess(String id, Object objectUserManager, String attributes,
+                               String excludeAttributes, int expectedScimResponseStatus, Object objectUser)
+                               throws BadRequestException, NotFoundException, CharonException {
+
+        UserManager userManager = (UserManager) objectUserManager;
+        User user = (User) objectUser;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema)
+                        CopyUtil.deepCopy(schema), attributes, excludeAttributes);
```

Comment:
No need to go to new line. Doesn't exceed 120 characters.

Reply:
fixed in 46de98fc2140524b63a2fc404dda55b17540ecce